<a href="https://colab.research.google.com/github/Nunticha24/data_viz_2022/blob/main/Project_Dinosaur.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Present Project กลุ่ม

https://drive.google.com/drive/u/0/folders/1G9XaOQ1IExr0m-UwWypqGPtGzi5HE-YL

In [57]:
import pandas as pd
import os

In [58]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [59]:
path = '/content/drive/My Drive/Project_Dinosaur'

#data covid

##ตารางข้อมูล covid ปี2020

* No รหัสผู้ป่วย
* Age อายุของผู้ป่วย
* Sex เพศผู้ป่วย
* Nationality สัญชาติ
* Province of isolation จังหวัดที่รักษาตัว
* Notification Date วันที่กระทรวงได้รับรายงาน
* Announce Date วันที่แถลงข่าว
* Province of onset จังหวัดที่เริ่มพบว่าตัวเองมีอาการป่วย
* District of onset อำเภอที่เริ่มพบว่าตัวเองมีอาการป่วย
* risk สถานที่

In [56]:
data_covid2020 =  pd.read_csv(os.path.join(path,'confirmed-cases.csv'))
print(data_covid2020.shape)
data_covid2020.head()

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


(839771, 11)


No. announce_date Notified date   sex   age Unit nationality  \
0    1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1    2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2    3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3    4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4    5     24/1/2020           NaN  หญิง  66.0   ปี       China   

  province_of_isolation                              risk province_of_onset  \
0         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4               นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   

  district_of_onset  
0               NaN  
1               NaN  
2             เมือง  
3               NaN  
4               NaN

##ตารางข้อมูล covid ปี2021(1)

In [60]:
data_covid2021_1 =  pd.read_csv(os.path.join(path,'confirmed-cases-since-120864.csv'))
print(data_covid2021_1.shape)
data_covid2021_1.head()

(1042168, 11)


No. announce_date Notified date   sex   age Unit nationality  \
0  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
1  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
2  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
3  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
4  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0              เชียงราย                                        อื่นๆ   
1              เชียงราย                                        อื่นๆ   
2              เชียงราย                                        อื่นๆ   
3              เชียงราย                                        อื่นๆ   
4              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN

##ตารางข้อมูล covid ปี2021(2)

In [61]:
data_covid2021_2 =  pd.read_csv(os.path.join(path,'confirmed-cases-since-271064.csv'))
print(data_covid2021_2.shape)
data_covid2021_2.head()

(1010459, 12)


/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


No. announce_date Notified date   sex   age Unit nationality  \
0  1859158    26/10/2021    25/10/2021   ชาย  20.0   ปี    Thailand   
1  1859159    26/10/2021    25/10/2021  หญิง  42.0   ปี    Thailand   
2  1859160    26/10/2021    25/10/2021  หญิง  33.0   ปี    Thailand   
3  1859161    26/10/2021    25/10/2021  หญิง  52.0   ปี    Thailand   
4  1859162    26/10/2021    25/10/2021  หญิง  84.0   ปี    Thailand   

  province_of_isolation                                           risk  \
0                 ลำปาง                                          อื่นๆ   
1                 ลำปาง                                          อื่นๆ   
2                 ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
3                 ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
4                 ลำปาง    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  Unnamed: 11  
0             ลำปาง        เมืองลำปาง          NaN  
1             ลำปาง        เมืองลำปาง          NaN  
2             ลำปาง          ห้างฉัตร          NaN  
3             ลำปาง          ห้างฉัตร          NaN  
4             ลำปาง        เมืองลำปาง          NaN

##ตารางข้อมูล covid ปี2022

In [62]:
data_covid2022 =  pd.read_csv(os.path.join(path,'confirmed-cases-since-280265.csv'))
print(data_covid2022.shape)
data_covid2022.head()

(433553, 11)


No. announce_date Notified date   sex   age Unit nationality  \
0  2869617     28/2/2022     27/2/2022   ชาย  41.0   ปี    Thailand   
1  2869618     28/2/2022     27/2/2022  หญิง  45.0   ปี    Thailand   
2  2869619     28/2/2022     27/2/2022  หญิง  40.0   ปี    Thailand   
3  2869620     28/2/2022     27/2/2022   ชาย  44.0   ปี    Thailand   
4  2869621     28/2/2022     27/2/2022   ชาย  11.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
1                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
2                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
3                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
4                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0             พะเยา        เมืองพะเยา  
1             พะเยา        เมืองพะเยา  
2             พะเยา        เมืองพะเยา  
3             พะเยา        เมืองพะเยา  
4             พะเยา        เมืองพะเยา

###ตรวจสอบว่าข้อมูลซ้ำกันหรือไม่

าค่า min max ของ column NO.

In [63]:
def min_max(i):
    a = min(i['No.'])
    b = max(i['No.'])
    print(f'min of NO. is {a}')
    print(f'max of NO. is {b}')

ดูค่า min max ของ covid ทั้ง 4 ตาราง

In [64]:
min_max(data_covid2020)

min of NO. is 1
max of NO. is 839771


In [65]:
min_max(data_covid2021_1)

min of NO. is 816990
max of NO. is 1859157


In [66]:
min_max(data_covid2021_2)

min of NO. is 1859158
max of NO. is 2869616


In [67]:
min_max(data_covid2022)

min of NO. is 2869617
max of NO. is 3303169


เห็นได้ว่าลักษณะของข้อมูลที่ดูจากค่าในคอลัมน์ NO. ของตาราง data_covid2020 และ data_covid2021_1 มีค่าซ้ำกัน

###ดูลักษณะข้อมูลว่าซ้ำกันจริงหรือไม่

In [68]:
data_covid2020.iloc[816989:,:].head(10)    #ดูข้อมูล data_covid2020 ตั้งแต่แถวที่ 816989 ถึงแถวสุดท้าย

No. announce_date Notified date   sex   age Unit nationality  \
816989  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
816990  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
816991  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
816992  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
816993  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
816994  816995     12/8/2021     11/8/2021   ชาย  39.0   ปี    Thailand   
816995  816996     12/8/2021     11/8/2021   ชาย  45.0   ปี    Thailand   
816996  816997     12/8/2021     11/8/2021  หญิง  32.0   ปี    Thailand   
816997  816998     12/8/2021     11/8/2021  หญิง   7.0   ปี    Thailand   
816998  816999     12/8/2021     11/8/2021   ชาย   8.0   ปี    Thailand   

       province_of_isolation                                         risk  \
816989              เชียงราย                                        อื่นๆ   
816990              เชียงราย                                        อื่นๆ   
816991              เชียงราย                                        อื่นๆ   
816992              เชียงราย                                        อื่นๆ   
816993              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
816994              เชียงราย                                        อื่นๆ   
816995              เชียงราย                                        อื่นๆ   
816996              เชียงราย                                        อื่นๆ   
816997              เชียงราย                                        อื่นๆ   
816998              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

       province_of_onset district_of_onset  
816989               NaN               NaN  
816990               NaN               NaN  
816991               NaN               NaN  
816992               NaN               NaN  
816993               NaN               NaN  
816994               NaN               NaN  
816995               NaN               NaN  
816996               NaN               NaN  
816997               NaN               NaN  
816998               NaN               NaN

In [69]:
data_covid2021_1.head(10)    #ดูข้อมูล data_covid2021_1

No. announce_date Notified date   sex   age Unit nationality  \
0  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
1  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
2  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
3  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
4  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
5  816995     12/8/2021     11/8/2021   ชาย  39.0   ปี    Thailand   
6  816996     12/8/2021     11/8/2021   ชาย  45.0   ปี    Thailand   
7  816997     12/8/2021     11/8/2021  หญิง  32.0   ปี    Thailand   
8  816998     12/8/2021     11/8/2021  หญิง   7.0   ปี    Thailand   
9  816999     12/8/2021     11/8/2021   ชาย   8.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0              เชียงราย                                        อื่นๆ   
1              เชียงราย                                        อื่นๆ   
2              เชียงราย                                        อื่นๆ   
3              เชียงราย                                        อื่นๆ   
4              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
5              เชียงราย                                        อื่นๆ   
6              เชียงราย                                        อื่นๆ   
7              เชียงราย                                        อื่นๆ   
8              เชียงราย                                        อื่นๆ   
9              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN  
5               NaN               NaN  
6               NaN               NaN  
7               NaN               NaN  
8               NaN               NaN  
9               NaN               NaN

จากการดูลักษณะข้อมูล จะเห็นว่าข้อมูลซ้ำกันจริง

จึงทำการลบข้อมูลที่ซ้ำออก โดยเลือกจากตาราง data_covid2020

In [70]:
data_covid2020_drop = data_covid2020.iloc[:816989,:]
data_covid2020_drop.head()

No. announce_date Notified date   sex   age Unit nationality  \
0    1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1    2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2    3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3    4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4    5     24/1/2020           NaN  หญิง  66.0   ปี       China   

  province_of_isolation                              risk province_of_onset  \
0         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4               นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   

  district_of_onset  
0               NaN  
1               NaN  
2             เมือง  
3               NaN  
4               NaN

In [71]:
min_max(data_covid2020_drop)   #ดู No. น้อยสุดและมากสุด ใน data_covid2020_drop

min of NO. is 1
max of NO. is 816989


In [72]:
min_max(data_covid2022) #ดู No. น้อยสุดและมากสุด ใน data_covid2022

min of NO. is 2869617
max of NO. is 3303169


บวกดูจำนวนข้อมูลของทั้ง 4 ตาราง

In [73]:
data_covid2020_drop.shape[0] + data_covid2021_1.shape[0] + data_covid2021_2.shape[0] + data_covid2022.shape[0]

3303169

จะเห็นว่าจำนวน NO. ที่มากที่สุดของข้อมูล เท่ากับจำนวนข้อมูลของทั้ง 4 ตาราง บวกกัน ซึ่งเท่ากับ 3303169 แสดงว่าข้อมูลแต่ละตารางไม่ได้เก็บซ้ำกันแล้ว

##รวมตาราง data_covid ทั้ง 4 ตาราง

In [74]:
data_covid = pd.concat([data_covid2020_drop, data_covid2021_1, data_covid2021_2, data_covid2022])
print(data_covid.shape)
data_covid.head()

(3303169, 12)


No. announce_date Notified date   sex   age Unit nationality  \
0    1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1    2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2    3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3    4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4    5     24/1/2020           NaN  หญิง  66.0   ปี       China   

  province_of_isolation                              risk province_of_onset  \
0         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4               นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   

  district_of_onset  Unnamed: 11  
0               NaN          NaN  
1               NaN          NaN  
2             เมือง          NaN  
3               NaN          NaN  
4               NaN          NaN

###เลือกเอา column ที่ต้องการ

In [75]:
covid = data_covid[['announce_date','province_of_isolation']]
print(covid.shape)
covid.head()

(3303169, 2)


announce_date province_of_isolation
0     12/1/2020         กรุงเทพมหานคร
1     17/1/2020         กรุงเทพมหานคร
2     22/1/2020                นครปฐม
3     22/1/2020         กรุงเทพมหานคร
4     24/1/2020               นนทบุรี

###Clean data_covid

In [76]:
covid.isnull().sum() #นับจำนวนค่า NaN ของแต่ละ columns

announce_date               0
province_of_isolation    3956
dtype: int64

###จัดรูปแบบข้อมูล

In [77]:
len(set(covid['province_of_isolation'])) #ข้อมูลในคอลัมน์ province_of_isolation ต้องมี 77 ค่า

81

In [78]:
set(covid['province_of_isolation'])

{nan,
 'กทม',
 'กระบี่',
 'กรุงเทพมหานคร',
 'กาญจนบุรี',
 'กาฬสินธุ์',
 'กำแพงเพชร',
 'ขอนแก่น',
 'ขอนแก่น\xa0',
 'จันทบุรี',
 'ฉะเชิงเทรา',
 'ชลบุรี',
 'ชัยนาท',
 'ชัยภูมิ',
 'ชุมพร',
 'ตรัง',
 'ตราด',
 'ตาก',
 'นครนายก',
 'นครปฐม',
 'นครพนม',
 'นครราชสีมา',
 'นครศรีธรรมราช',
 'นครสวรรค์',
 'นนทบุรี',
 'นราธิวาส',
 'น่าน',
 'บึงกาฬ',
 'บุรีรัมย์',
 'ปทุมธานี',
 'ประจวบคีรีขันธ์',
 'ปราจีนบุรี',
 'ปัตตานี',
 'พระนครศรีอยุธยา',
 'พะเยา',
 'พังงา',
 'พัทลุง',
 'พิจิตร',
 'พิษณุโลก',
 'ภููเก็ต',
 'ภูเก็ต',
 'มหาสารคาม',
 'มุกดาหาร',
 'ยะลา',
 'ยโสธร',
 'ระนอง',
 'ระยอง',
 'ราชบุรี',
 'ร้อยเอ็ด',
 'ลพบุรี',
 'ลำปาง',
 'ลำพูน',
 'ศรีสะเกษ',
 'สกลนคร',
 'สงขลา',
 'สตูล',
 'สมุทรปราการ',
 'สมุทรสงคราม',
 'สมุทรสาคร',
 'สระบุรี',
 'สระแก้ว',
 'สิงห์บุรี',
 'สุพรรณบุรี',
 'สุราษฎร์ธานี',
 'สุรินทร์',
 'สุโขทัย',
 'หนองคาย',
 'หนองบัวลำภู',
 'อำนาจเจริญ',
 'อุดรธานี',
 'อุตรดิตถ์',
 'อุทัยธานี',
 'อุบลราชธานี',
 'อ่างทอง',
 'เชียงราย',
 'เชียงใหม่',
 'เพชรบุรี',
 'เพชรบูรณ์',
 'เลย',
 'แพร่',
 '

In [79]:
covid['province_of_isolation'].replace({'กทม':'กรุงเทพมหานคร','ขอนแก่น\xa0':'ขอนแก่น','ภููเก็ต':'ภูเก็ต'}, inplace=True)

/usr/local/lib/python3.7/dist-packages/pandas/core/generic.py:6619: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return self._update_inplace(result)


In [80]:
len(set(covid['province_of_isolation'])) #มีค่าที่ชื่อว่า ไม่ระบุ

78

In [81]:
covid.isnull().sum() #นับจำนวนค่า NaN ของแต่ละ columns

announce_date               0
province_of_isolation    3956
dtype: int64

In [82]:
covid_dropped = covid.dropna() #drop ค่า Nan ออกจากตาราง

In [83]:
covid_dropped.isnull().sum()

announce_date            0
province_of_isolation    0
dtype: int64

In [84]:
print(covid_dropped.shape) #ตารางสุดท้ายพร้อมใช้งานของ data covid
covid_dropped.head()

(3299213, 2)


announce_date province_of_isolation
0     12/1/2020         กรุงเทพมหานคร
1     17/1/2020         กรุงเทพมหานคร
2     22/1/2020                นครปฐม
3     22/1/2020         กรุงเทพมหานคร
4     24/1/2020               นนทบุรี

#data region

In [85]:
data_region = {'Province':['อำนาจเจริญ','บึงกาฬ','บุรีรัมย์','ชัยภูมิ','กาฬสินธุ์','ขอนแก่น','เลย','มหาสารคาม','มุกดาหาร','นครพนม','นครราชสีมา','หนองบัวลำภู','หนองคาย','ร้อยเอ็ด','สกลนคร','ศรีสะเกษ','สุรินทร์','อุบลราชธานี','อุดรธานี','ยโสธร','เชียงใหม่','เชียงราย','ลำปาง','ลำพูน','แม่ฮ่องสอน','น่าน','พะเยา','แพร่','อุตรดิตถ์','ตาก','สุโขทัย','พิษณุโลก','พิจิตร','กำแพงเพชร','เพชรบูรณ์','นครสวรรค์','อุทัยธานี','อ่างทอง','ชัยนาท','พระนครศรีอยุธยา','กรุงเทพมหานคร','ลพบุรี','นครปฐม','นนทบุรี','ปทุมธานี','สมุทรปราการ','สมุทรสาคร','สมุทรสงคราม','สระบุรี','สิงห์บุรี','สุพรรณบุรี','นครนายก','ฉะเชิงเทรา','จันทบุรี','ชลบุรี','ปราจีนบุรี','ระยอง','สระแก้ว','ตราด','กาญจนบุรี','ราชบุรี','เพชรบุรี','ประจวบคีรีขันธ์','ชุมพร','นครศรีธรรมราช','นราธิวาส','ปัตตานี','พัทลุง','สงขลา','สุราษฎร์ธานี','ยะลา','กระบี่','พังงา','ภูเก็ต','ระนอง','สตูล','ตรัง'],
             '6 ภูมิภาค':['ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','ตะวันตก','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันตก','ตะวันตก','ตะวันตก','ตะวันตก','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้'],
}

In [86]:
data_region = pd.DataFrame.from_dict(data_region)
print(data_region.shape)
data_region.head()

(77, 2)


Province           6 ภูมิภาค
0  อำนาจเจริญ  ตะวันออกเฉียงเหนือ
1      บึงกาฬ  ตะวันออกเฉียงเหนือ
2   บุรีรัมย์  ตะวันออกเฉียงเหนือ
3     ชัยภูมิ  ตะวันออกเฉียงเหนือ
4   กาฬสินธุ์  ตะวันออกเฉียงเหนือ

In [87]:
data_region.isnull().sum()

Province     0
6 ภูมิภาค    0
dtype: int64

In [88]:
print(data_region.shape) #ตารางสุดท้ายพร้อมใช้งานของ data region
data_region.head()

(77, 2)


Province           6 ภูมิภาค
0  อำนาจเจริญ  ตะวันออกเฉียงเหนือ
1      บึงกาฬ  ตะวันออกเฉียงเหนือ
2   บุรีรัมย์  ตะวันออกเฉียงเหนือ
3     ชัยภูมิ  ตะวันออกเฉียงเหนือ
4   กาฬสินธุ์  ตะวันออกเฉียงเหนือ

#นำเข้าข้อมูลที่เลือกเอง
การเดินทางของประชาชนในช่วงการแพร่ระบาดของเชื้อไวรัส COVID-19
ที่มา : https://datagov.mot.go.th/dataset/covid-19 (กระทรวงคมนาคม)

In [89]:
passenger =  pd.read_csv(os.path.join(path,'passenger-covid19.csv'))
print(passenger.shape)
passenger.head()

(810, 28)


วันที่ ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
0  1/1/2020   1,483,255     885,353     272,578        102,678   
1  2/1/2020   1,705,026   1,269,207     291,990        104,613   
2  3/1/2020   1,752,572   1,366,412     301,260        100,575   
3  4/1/2020   1,576,146   1,027,281     271,171        100,852   
4  5/1/2020   1,565,134     915,425     254,176        100,784   

  รวมประชาชนเดินทาง (คน) รถส่วนบุคคล เข้า - ออก กทม. (คัน)  \
0              2,743,864                         2,310,406   
1              3,370,836                         2,538,456   
2              3,520,819                         2,649,091   
3              2,975,450                         2,515,920   
4              2,835,519                         2,542,483   

  รถโดยสารประจำทางใน กทม. และปริมณฑล (ขสมก. และ รถร่วม) (คน)  \
0                                          1,249,925           
1                                          1,452,127           
2                                          1,526,841           
3                                          1,359,227           
4                                          1,330,523           

  รถโดยสารประจำทางระหว่าง กทม. - ตจว. \n(บขส. และ รถร่วม) (คน)  \
0                                            233,330             
1                                            252,899             
2                                            225,731             
3                                            216,919             
4                                            234,611             

  รถส่วนบุคคล เข้า - ออก กทม. (คัน).1  ... รถไฟระหว่างเมือง เรือด่วนเจ้าพระยา  \
0                           2,310,406  ...          115,975            29,518   
1                           2,538,456  ...          103,113            30,794   
2                           2,649,091  ...           98,657            31,036   
3                           2,515,920  ...           94,876            20,236   
4                           2,542,483  ...           98,259            17,066   

  เรือคลองแสนแสบ เรือข้ามฟากเจ้าพระยา เรือไฟฟ้าเจ้าพระยา เรือในภูมิภาค  \
0         20,420                    0                  0       222,640   
1         35,430                    0                  0       225,766   
2         32,540                    0                  0       237,684   
3         20,850                    0                  0       230,085   
4         20,030                    0                  0       217,080   

  รวมท่าอากาศยานใน กทม. ท่าอากาศยานสุวรรณภูมิ ท่าอากาศยานดอนเมือง  \
0                45,774                14,674              31,100   
1                45,045                14,974              30,071   
2                41,894                13,036              28,858   
3                40,186                12,163              28,023   
4                40,161                12,468              27,693   

  ท่าอากาศยานภูมิภาค  
0             56,904  
1             59,568  
2             58,681  
3             60,666  
4             60,623  

[5 rows x 28 columns]

In [90]:
Passenger = passenger[['วันที่','ทางถนน (คน)','ทางราง (คน)','ทางน้ำ (คน)', 'ทางอากาศ* (คน)', 'รวมประชาชนเดินทาง (คน)']]
print(Passenger.shape)
Passenger.head()

(810, 6)


วันที่ ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
0  1/1/2020   1,483,255     885,353     272,578        102,678   
1  2/1/2020   1,705,026   1,269,207     291,990        104,613   
2  3/1/2020   1,752,572   1,366,412     301,260        100,575   
3  4/1/2020   1,576,146   1,027,281     271,171        100,852   
4  5/1/2020   1,565,134     915,425     254,176        100,784   

  รวมประชาชนเดินทาง (คน)  
0              2,743,864  
1              3,370,836  
2              3,520,819  
3              2,975,450  
4              2,835,519

In [91]:
Passenger.isnull().sum() #เช็คค่าว่าง

วันที่                    0
ทางถนน (คน)               0
ทางราง (คน)               0
ทางน้ำ (คน)               0
ทางอากาศ* (คน)            0
รวมประชาชนเดินทาง (คน)    0
dtype: int64

In [92]:
print(Passenger.shape) #ตารางพร้อมใช้งานของ passenger
Passenger.head()

(810, 6)


วันที่ ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
0  1/1/2020   1,483,255     885,353     272,578        102,678   
1  2/1/2020   1,705,026   1,269,207     291,990        104,613   
2  3/1/2020   1,752,572   1,366,412     301,260        100,575   
3  4/1/2020   1,576,146   1,027,281     271,171        100,852   
4  5/1/2020   1,565,134     915,425     254,176        100,784   

  รวมประชาชนเดินทาง (คน)  
0              2,743,864  
1              3,370,836  
2              3,520,819  
3              2,975,450  
4              2,835,519

##รวมตาราง

ชื่อข้อมูลตารางที่พร้อมใช้งาน
* covid_dropped
* data_region
* Passenger



###ต่อตาราง covid ด้วยตาราง region

In [93]:
covid_region = covid_dropped.merge(data_region,how='left',left_on='province_of_isolation',right_on='Province') 
print(covid_region.shape)
covid_region.head(3)

(3299213, 4)


announce_date province_of_isolation       Province 6 ภูมิภาค
0     12/1/2020         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง
1     17/1/2020         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง
2     22/1/2020                นครปฐม         นครปฐม      กลาง

###ต่อตาราง covid ด้วยตาราง region

In [94]:
all_data = covid_region.merge(Passenger,how='left',left_on='announce_date',right_on='วันที่')     # ต่อตาราง covid ด้วยตาราง region
print(all_data.shape)
all_data.head(3)

(3299213, 10)


announce_date province_of_isolation       Province 6 ภูมิภาค     วันที่  \
0     12/1/2020         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง  12/1/2020   
1     17/1/2020         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง  17/1/2020   
2     22/1/2020                นครปฐม         นครปฐม      กลาง  22/1/2020   

  ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน) รวมประชาชนเดินทาง (คน)  
0   1,511,236     988,051     246,025         99,755              2,845,067  
1   1,732,040   1,599,647     281,743        106,000              3,719,430  
2   1,678,182   1,488,097     267,105        101,432              3,534,816

In [95]:
all_data.isnull().any()

announce_date             False
province_of_isolation     False
Province                  False
6 ภูมิภาค                 False
วันที่                    False
ทางถนน (คน)               False
ทางราง (คน)               False
ทางน้ำ (คน)               False
ทางอากาศ* (คน)            False
รวมประชาชนเดินทาง (คน)    False
dtype: bool

#เปลี่ยนชนิดของข้อมูล

In [96]:
all_data['announce_date'] = pd.to_datetime(all_data['announce_date'],format="%d/%m/%Y")

In [97]:
all_data.dtypes

announce_date             datetime64[ns]
province_of_isolation             object
Province                          object
6 ภูมิภาค                         object
วันที่                            object
ทางถนน (คน)                       object
ทางราง (คน)                       object
ทางน้ำ (คน)                       object
ทางอากาศ* (คน)                    object
รวมประชาชนเดินทาง (คน)            object
dtype: object

In [98]:
all_data

announce_date province_of_isolation       Province 6 ภูมิภาค  \
0          2020-01-12         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง   
1          2020-01-17         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง   
2          2020-01-22                นครปฐม         นครปฐม      กลาง   
3          2020-01-22         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง   
4          2020-01-24               นนทบุรี        นนทบุรี      กลาง   
...               ...                   ...            ...       ...   
3299208    2022-03-18         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง   
3299209    2022-03-18         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง   
3299210    2022-03-18         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง   
3299211    2022-03-18                 สงขลา          สงขลา       ใต้   
3299212    2022-03-18                 สงขลา          สงขลา       ใต้   

            วันที่ ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
0        12/1/2020   1,511,236     988,051     246,025         99,755   
1        17/1/2020   1,732,040   1,599,647     281,743        106,000   
2        22/1/2020   1,678,182   1,488,097     267,105        101,432   
3        22/1/2020   1,678,182   1,488,097     267,105        101,432   
4        24/1/2020   1,658,692   1,527,329     281,291        103,697   
...            ...         ...         ...         ...            ...   
3299208  18/3/2022     646,203     729,250     104,801         63,230   
3299209  18/3/2022     646,203     729,250     104,801         63,230   
3299210  18/3/2022     646,203     729,250     104,801         63,230   
3299211  18/3/2022     646,203     729,250     104,801         63,230   
3299212  18/3/2022     646,203     729,250     104,801         63,230   

        รวมประชาชนเดินทาง (คน)  
0                    2,845,067  
1                    3,719,430  
2                    3,534,816  
3                    3,534,816  
4                    3,571,009  
...                        ...  
3299208              1,543,484  
3299209              1,543,484  
3299210              1,543,484  
3299211              1,543,484  
3299212              1,543,484  

[3299213 rows x 10 columns]

#แยกข้อมูล
เนื่องจากขนาดไฟล์ใหญ่เกินไปจึงแยกข้อมูลออกเป็นส่วนๆ

In [99]:
covid2020 = all_data[all_data['announce_date'] < '2021-01-01']  #เลือกข้อมูลตั้งแต่วันแรก ถึง วันที่ 2020-12-31
print(covid2020.shape)                                            
covid2020.head(3)

(6862, 10)


announce_date province_of_isolation       Province 6 ภูมิภาค     วันที่  \
0    2020-01-12         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง  12/1/2020   
1    2020-01-17         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง  17/1/2020   
2    2020-01-22                นครปฐม         นครปฐม      กลาง  22/1/2020   

  ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน) รวมประชาชนเดินทาง (คน)  
0   1,511,236     988,051     246,025         99,755              2,845,067  
1   1,732,040   1,599,647     281,743        106,000              3,719,430  
2   1,678,182   1,488,097     267,105        101,432              3,534,816

In [100]:
covid2021_1 = all_data[(all_data['announce_date'] >= '2021-01-01') & (all_data['announce_date'] < '2021-04-01')] #เลือกข้อมูลไตรมาสแรกของปี 2021
print(covid2021_1.shape)
covid2021_1.head(3)

(21870, 10)


announce_date province_of_isolation       Province 6 ภูมิภาค    วันที่  \
120     2021-03-06              ปทุมธานี       ปทุมธานี      กลาง  6/3/2021   
6863    2021-01-01         กรุงเทพมหานคร  กรุงเทพมหานคร      กลาง  1/1/2021   
6864    2021-01-01              ปทุมธานี       ปทุมธานี      กลาง  1/1/2021   

     ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน) รวมประชาชนเดินทาง (คน)  
120      862,101     682,272      98,061         47,995              1,690,429  
6863     629,297     331,738     132,198         53,527              1,146,760  
6864     629,297     331,738     132,198         53,527              1,146,760

In [101]:
covid2021_2 = all_data[(all_data['announce_date'] >= '2021-04-01') & (all_data['announce_date'] < '2021-07-01')] #เลือกข้อมูลไตรมาสสองของปี 2021
print(covid2021_2.shape)
covid2021_2.head(3)

(230297, 10)


announce_date province_of_isolation   Province 6 ภูมิภาค    วันที่  \
28732    2021-04-01             สมุทรสาคร  สมุทรสาคร      กลาง  1/4/2021   
28733    2021-04-01             สมุทรสาคร  สมุทรสาคร      กลาง  1/4/2021   
28734    2021-04-01             สมุทรสาคร  สมุทรสาคร      กลาง  1/4/2021   

      ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
28732   1,002,983   1,037,660     102,982         70,353   
28733   1,002,983   1,037,660     102,982         70,353   
28734   1,002,983   1,037,660     102,982         70,353   

      รวมประชาชนเดินทาง (คน)  
28732              2,213,978  
28733              2,213,978  
28734              2,213,978

In [102]:
covid2021_3 = all_data[(all_data['announce_date'] >= '2021-07-01') & (all_data['announce_date'] < '2021-10-01')] #เลือกข้อมูลไตรมาสสามของปี 2021
print(covid2021_3.shape)
covid2021_3.head(3)

(1344174, 10)


announce_date province_of_isolation  Province 6 ภูมิภาค    วันที่  \
259029    2021-07-01              เชียงราย  เชียงราย     เหนือ  1/7/2021   
259030    2021-07-01              เชียงราย  เชียงราย     เหนือ  1/7/2021   
259031    2021-07-01              เชียงราย  เชียงราย     เหนือ  1/7/2021   

       ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
259029     574,401     447,352      50,212         13,123   
259030     574,401     447,352      50,212         13,123   
259031     574,401     447,352      50,212         13,123   

       รวมประชาชนเดินทาง (คน)  
259029              1,085,088  
259030              1,085,088  
259031              1,085,088

In [103]:
covid2021_4 = all_data[(all_data['announce_date'] >= '2021-10-01') & (all_data['announce_date'] < '2022-01-01')] #เลือกข้อมูลไตรมาสสี่ของปี 2021
print(covid2021_4.shape)
covid2021_4.head(3)

(616276, 10)


announce_date province_of_isolation  Province 6 ภูมิภาค     วันที่  \
1603203    2021-10-01                 พะเยา     พะเยา     เหนือ  1/10/2021   
1603204    2021-10-01                 พะเยา     พะเยา     เหนือ  1/10/2021   
1603205    2021-10-01              เชียงราย  เชียงราย     เหนือ  1/10/2021   

        ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
1603203     542,326     485,393      51,797         19,117   
1603204     542,326     485,393      51,797         19,117   
1603205     542,326     485,393      51,797         19,117   

        รวมประชาชนเดินทาง (คน)  
1603203              1,098,633  
1603204              1,098,633  
1603205              1,098,633

In [105]:
covid2022_1 = all_data[(all_data['announce_date'] >= '2022-01-01') & (all_data['announce_date'] < '2022-02-01')] #เลือกข้อมูลเดือนมกราคม 2022
print(covid2022_1.shape)
covid2022_1.head(3)

(217107, 10)


announce_date province_of_isolation Province 6 ภูมิภาค    วันที่  \
2219479    2022-01-01                 พะเยา    พะเยา     เหนือ  1/1/2022   
2219480    2022-01-01                 พะเยา    พะเยา     เหนือ  1/1/2022   
2219481    2022-01-01                 พะเยา    พะเยา     เหนือ  1/1/2022   

        ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
2219479     515,514     411,352     151,399         56,701   
2219480     515,514     411,352     151,399         56,701   
2219481     515,514     411,352     151,399         56,701   

        รวมประชาชนเดินทาง (คน)  
2219479              1,134,966  
2219480              1,134,966  
2219481              1,134,966

In [106]:
covid2022_2 = all_data[(all_data['announce_date'] >= '2022-02-01') & (all_data['announce_date'] < '2022-03-01')] #เลือกข้อมูลเดือนกุมภาพันธ์ 2022
print(covid2022_2.shape)
covid2022_2.head(3)

(451385, 10)


announce_date province_of_isolation Province 6 ภูมิภาค    วันที่  \
2436586    2022-02-01                 พะเยา    พะเยา     เหนือ  1/2/2022   
2436587    2022-02-01                 พะเยา    พะเยา     เหนือ  1/2/2022   
2436588    2022-02-01                 พะเยา    พะเยา     เหนือ  1/2/2022   

        ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
2436586     714,901     745,660      99,446         38,761   
2436587     714,901     745,660      99,446         38,761   
2436588     714,901     745,660      99,446         38,761   

        รวมประชาชนเดินทาง (คน)  
2436586              1,598,768  
2436587              1,598,768  
2436588              1,598,768

In [109]:
covid2022_3 = all_data[all_data['announce_date'] >= '2022-03-01'] #เลือกข้อมูลเดือนมีนาคม 2022 ถึงล่าสุดที่โหลดมา
print(covid2022_3.shape)
covid2022_3.head(3)

(411242, 10)


announce_date province_of_isolation    Province 6 ภูมิภาค    วันที่  \
2887971    2022-03-01            แม่ฮ่องสอน  แม่ฮ่องสอน     เหนือ  1/3/2022   
2887972    2022-03-01            แม่ฮ่องสอน  แม่ฮ่องสอน     เหนือ  1/3/2022   
2887973    2022-03-01            แม่ฮ่องสอน  แม่ฮ่องสอน     เหนือ  1/3/2022   

        ทางถนน (คน) ทางราง (คน) ทางน้ำ (คน) ทางอากาศ* (คน)  \
2887971     666,637     696,843      83,386         41,226   
2887972     666,637     696,843      83,386         41,226   
2887973     666,637     696,843      83,386         41,226   

        รวมประชาชนเดินทาง (คน)  
2887971              1,488,092  
2887972              1,488,092  
2887973              1,488,092

#โหลดข้อมูล

In [111]:
covid2020.to_csv('covid2020.csv',index=False) #โหลดข้อมูลตั้งแต่วันแรก ถึง วันที่ 2020-12-31
covid2021_1.to_csv('covid2021_1.csv',index=False) #โหลดข้อมูลไตรมาสแรกของปี 2021
covid2021_2.to_csv('covid2021_2.csv',index=False) #โหลดข้อมูลไตรมาสสองของปี 2021
covid2021_3.to_csv('covid2021_3.csv',index=False) #โหลดข้อมูลไตรมาสสามของปี 2021
covid2021_4.to_csv('covid2021_4.csv',index=False)#โหลดข้อมูลไตรมาสสี่ของปี 2021
covid2022_1.to_csv('covid2022_1.csv',index=False) #โหลดข้อมูลเดือนมกราคม 2022
covid2022_2.to_csv('covid2022_2.csv',index=False) #โหลดข้อมูลเดือนกุมภาพันธ์ 2022
covid2022_3.to_csv('covid2022_3.csv',index=False) #โหลดข้อมูลเดือนมีนาคม 2022 ถึงล่าสุดที่โหลดมา